packages/notifications/__tests__/testUtils/mergeInAppMessageWithOverrides.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { cloneDeep } from "lodash";
+import { InAppMessageCampaign, OverrideButtonConfiguration } from "@aws-amplify/core/internals/aws-clients/pinpoint";
+import { InAppMessageButton, InAppMessageContent } from "../../src/inAppMessaging/types/message";
+import { mapOSPlatform } from "../../src/inAppMessaging/providers/pinpoint/utils/helpers";
+
+export const mergeInAppMessageWithOverrides:(
+	pinpointInAppMessage:InAppMessageCampaign,
+	platform:'android'|'ios'|'web',
+	buttonOverrides?:{
+		primaryButton:OverrideButtonConfiguration,
+		secondaryButton:OverrideButtonConfiguration
+	}
+ ) =>InAppMessageCampaign = (pinpointInAppMessage, platform, buttonOverrides)=>{
+	const message= cloneDeep(pinpointInAppMessage);
+	const configPlatform = mapOSPlatform(platform);
```

Comment:
Would highly prefer not relying on other utils inside a test util as it could be hard to debug if something upstream changes and causes test failures downstream. In this case, since the goal is to augment some `InAppMessageCampaign`'s button configuration with some `ButtonConfigPlatform` keyed values, I think we can just have the function accept that directly. We can also tidy up the signature a bit by typing the parameters inline: E.g.
```suggestion
export const mergeInAppMessageWithOverrides = (
	pinpointInAppMessage: InAppMessageCampaign,
	configPlatform: ButtonConfigPlatform,
	buttonOverrides?: {
		primaryButton: OverrideButtonConfiguration;
		secondaryButton: OverrideButtonConfiguration;
	}
): InAppMessageCampaign => {
	const message = cloneDeep(pinpointInAppMessage);
```

Reply:
I agree, but I see minimal risk in debugging, since I've added a test for the mappedOS utils. If any upstream changes affect this function, the test should fail, highlighting the issue. Regardless, to avoid using additional utils in the test utils, I updated the mock data to include a mappedPlatform property alongside configPlatform and passed it down. I'm open to suggestions.

[updated merge override fns signature and associated tests](https://github.com/aws-amplify/amplify-js/pull/12917/commits/3a58f157071dcb7ee62a9e8bf062c96bb24d7cd0)